src/Confluent.Kafka/KafkaException.cs

```diff
@@ -40,6 +40,22 @@ public KafkaException(Error error)
         }
 
         /// <summary>
+        ///     Initialize a new instance of KafkaException based on
+        ///     an existing Error instance and inner exception.
+        /// </summary>
+        /// <param name="error">
+        ///     The Kafka Error.
+        /// </param>
+        /// <param name="innerException">
+        ///     The exception instance that caused this exception.
+        /// </param>
+        public KafkaException(Error error, Exception innerException)
```

Comment:
The InnerException is never used (because the exception is eventually squashed down into an `Error`), but I thought we might as well propagate it whilst it's easy to do so. In practice, not propagating this all the way to the `OnConsumeError` event is not a problem I think, as a custom Message string will be enough in practice in this scenario.